### PR TITLE
[SPARK-10415][PySpark][MLlib][Docs] Enhance Navigation Sidebar in PySpark API

### PR DESCRIPTION
**Pull request by noel-smith**
These are CSS/JavaScript changes changes to make navigation in the PySpark API a bit simpler by adding the following to the sidebar:
- Classes
- Functions
- Tags to highlight experimental features

screen shot 2015-09-02 at 08 50 12

Online example here: https://dl.dropboxusercontent.com/u/20821334/pyspark-api-nav-enhance/pyspark.mllib.html

(The contribution is my original work and that I license the work to the project under the project's open source license)

### CODE REVIEW

**Comment by srowen** [comment]
Looks pretty simple to me. The change is visually appealing. It would be great to get a third opinion on that before proceeding

File: python/docs/_templates/layout.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{% extends "!layout.html" %}
+{% set script_files = script_files + ["_static/pyspark.js"] %}
+{% set css_files = css_files + ['_static/pyspark.css'] %}
+{% block rootrellink %}
+    {{ super() }}
+{% endblock %}
```
Comment (davies) [inline python/docs/_templates/layout.html]: new line here

**Comment by davies** [comment]
OK to test

File: python/docs/_static/pyspark.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
+body {
```
Comment (davies) [inline python/docs/_static/pyspark.css]: Add license header here.

**Comment by davies** [comment]
@noel-smith I tried this locally, after `$ make html` in python/docs, can't get the sidebar work, did I miss something?

**Comment by noel-smith** [comment]
Hmm that's odd - that's exactly the way I'm building too. Did your generated html pages include the links to the `pyspark.js` and `pyspark.css` files?

**Comment by davies** [comment]
@noel-smith Yes, I saw them in the generated html. There is a sidebar on the left, but all the titles are empty (but having E flag).

**Comment by noel-smith** [comment]
@davies Could you let me know your browser/platform + whether the example [link](https://dl.dropboxusercontent.com/u/20821334/pyspark-api-nav-enhance/pyspark.mllib.html) works for you? I'll try and reproduce locally.

**Comment by davies** [comment]
That link work well, I'm using Mac with Homebrew, should I update Sphinx ? Currently running v1.2.3

**Comment by davies** [comment]
After update Sphinx to 1.3.1, it worked! We may need to update this: `needs_sphinx = '1.3'`

**Comment by noel-smith** [comment]
Nice - thanks for finding that! 

It probably indicates I need to make the JS logic a bit more robust though. I'll take a look at the Sphinx history and find out how it's changed.

**Comment by noel-smith** [comment]
Tweaked JavaScript to make it more robust to Sphinx changes. Set minimal version of Sphinx to `1.2`.

**Comment by davies** [comment]
Merging this into master, thanks for the great job!

**Pull request closed without merging**